### PR TITLE
Task IDs parity between paper and code

### PR DESCRIPTION
**Pull request by MarcCote**
Currently, there is a mismatch between task IDs mentioned in the paper and the ones used in the code.

For instance, in the code "Measurement" task has ID 10-*,
```

> ScienceWorldEnv("").getTaskNames()
['task-1-boil',
 'task-1-change-the-state-of-matter-of',
 'task-1-freeze',
 'task-1-melt',
 'task-10-measure-melting-point-(known-substance)',      <---Mismatch
 'task-10-measure-melting-point-(unknown-substance)',    <---Mismatch
 'task-10-use-thermometer',                              <---Mismatch
 'task-2-power-component',
 'task-2-power-component-(renewable-vs-nonrenewable-energy)',
 'task-2a-test-conductivity',
 'task-2a-test-conductivity-of-unknown-substances',
 'task-3-find-animal',
 'task-3-find-living-thing',
...
```
and in the paper it is ID 2
image

This PR addresses this issue by removing the prefix "task-ID-" from the task names. Also, this PR changes the `env.load(...)` function to accept a proper task ID to load a task, e.g. `env.load("2-3")`.

Last, a Python script is available to list all task names, their IDs, and number of variations.
`python scripts/list_scienceworld_task.py`
```
TASK LIST:
   1-1                                                boil      (  30 variations)
   1-2                                                melt      (  30 variations)
   1-3                                              freeze      (  30 variations)
   1-4                       change-the-state-of-matter-of      (  30 variations)
   2-1                                     use-thermometer      ( 540 variations)
   2-2             measure-melting-point-(known-substance)      ( 436 variations)
   2-3           measure-melting-point-(unknown-substance)      ( 300 variations)
   3-1                                     power-component      (  20 variations)
   3-2  power-component-(renewable-vs-nonrenewable-energy)      (  20 variations)
   3-3                                   test-conductivity      ( 900 variations)
   3-4             test-conductivity-of-unknown-substances      ( 600 variations)
   4-1                                   find-living-thing      ( 300 variations)
   4-2                               find-non-living-thing      ( 300 variations)
   4-3                                          find-plant      ( 300 variations)
   4-4                                         find-animal      ( 300 variations)
   5-1                                          grow-plant      ( 126 variations)
   5-2                                          grow-fruit      ( 126 variations)
   6-1                                       chemistry-mix      (  32 variations)
   6-2               chemistry-mix-paint-(secondary-color)      (  36 variations)
   6-3                chemistry-mix-paint-(tertiary-color)      (  36 variations)
   7-1                            lifespan-(longest-lived)      ( 125 variations)
   7-2                           lifespan-(shortest-lived)      ( 125 variations)
   7-3        lifespan-(longest-lived-then-shortest-lived)      ( 125 variations)
   8-1                              identify-life-stages-1      (  14 variations)
   8-2                              identify-life-stages-2      (  10 variations)
   9-1                      inclined-plane-determine-angle      ( 168 variations)
   9-2            inclined-plane-friction-(named-surfaces)      (1386 variations)
   9-3          inclined-plane-friction-(unnamed-surfaces)      ( 162 variations)
  10-1                    mendelian-genetics-(known-plant)      ( 120 variations)
  10-2                  mendelian-genetics-(unknown-plant)      ( 480 variations)
```

### CODE REVIEW

File: scienceworld/constants.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    if env_value in {"1", "true"}:
+        return True
+    elif env_value in {"", "0", "false"}:
+        return False
+    else:
+        raise ValueError(f'{env_value!r} is not a valid value for "SCIENCEWORLD_DEBUG"')
+
+DEBUG_MODE = is_in_debug_mode()
+
+ID2TASK = OrderedDict([
```
[the remark this answers is not in the document]
Comment (MarcCote) [reply scienceworld/constants.py:24]: That's a good point. I had issues in the past with dict ordering but that was early on with Python 3.

I don't think we decided what version of Python we wanted to support. I'm fine with focusing on 3.8+.

**Comment by MarcCote** [comment]
@aphedges hopefully this PR is not breaking anything on your side. You can still use the former task name convention. Let me know if that's not the case.

File: scienceworld/constants.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (aphedges) [reply scienceworld/constants.py:24]: Thanks for clarifying! I might put out a PR to enforce it.

**Comment by aphedges** [comment]
> @aphedges hopefully this PR is not breaking anything on your side. You can still use the former task name convention. Let me know if that's not the case.

It might break some stuff, but we have enough internal breakage that we already have scripts to do task name translation.

ISI is currently on break, so it won't be something we will be discussing until January.

**Comment by MarcCote** [comment]
Once I'm done with https://github.com/allenai/ScienceWorld/pull/37, there will be some breaking changes too. I plan to make ScienceWorld compatible with Gymnasium.

**Pull request merged**